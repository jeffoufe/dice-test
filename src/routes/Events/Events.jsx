// @flow

import React, { useEffect } from 'react';
import { fetchEvents } from '../../reducers/events/actions';
import { withRouter, Redirect } from 'react-router-dom'
import qs from 'qs';
import { connect } from 'react-redux';
import EventCard from '../../components/EventCard/EventCard';
import type { EventsState } from '../../reducers/events/reducer';

type PureEventsProps = {
    location: {
        search: string
    },
    fetchEvents: (string) => void,
    events: EventsState
}


export const PureEvents = ({ location, fetchEvents, events }: PureEventsProps) => {
    const venue = qs.parse(location.search.substring(1)).venue;

    useEffect(() => {
        if (events.willFetchEvents) {
            fetchEvents(venue);
        }
    })

    return venue
        ? (
           <div>
               {!!events.events.length && <EventCard {...events.events[0]} />}
            </div>
        )
        : <Redirect to='/' />
}

const mapStateToProps = state => ({ events: state.events })
const mapDispatchToProps = dispatch => ({ fetchEvents: (venue: string) => dispatch(fetchEvents(venue) )})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(PureEvents));