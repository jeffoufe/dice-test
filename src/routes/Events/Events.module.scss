@import '../../constants/_breakpoints';

.events {
    &-loading {
        text-align: center;
        margin-top: 20px;
    }

    &-container {
        padding: 60px 30px;

        @media screen and (min-width: $tablet) {
            padding: 90px 0 0;
            display: flex;
            justify-content: center;
            flex-wrap: wrap;
            -webkit-box-pack: center;
        }
    }

    &-error {
        margin-top: 30px;
    }

    &-load-more {
        position: absolute;
        left: 50%;
        transform: translateX(-50%);
        margin-bottom: 40px;
    }
}