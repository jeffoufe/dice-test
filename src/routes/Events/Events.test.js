import { shallow } from 'enzyme';
import React from 'react';
import { PureEvents } from './Events';

jest.mock('qs', () => ({ parse: () => ({ venue: 'venue' })}));

describe('Events', () => {
    const event = {
        event_images: {
            landscape: 'landscape',
            square: 'square'
        },
        apple_music_tracks: [],
        spotify_tracks: [],
        ticket_types: [{ price: { total: 800 } }, { price: { total: 400 } }],
        featured: false,
        sale_start_date: '2019-08-19',
        date: 'date',
        name: 'name',
        venue: 'venue',
        description: 'description',
        sold_out: false,
        url: 'url',
        lineup: [],
        currency: 'GBP',
        location: {
            city: 'city',
            country: 'country'
        }
    }

    const props = {
        location: {
            search: 'search'
        },
        triggerFetch: jest.fn(),
        reset: jest.fn(),
        fetchEvents: jest.fn(),
        events: {
            events: [event, event], 
            willFetchEvents: false, 
            isLoading: false, 
            links: {}, 
            error: ''
        }
    }

    const wrapper = shallow(<PureEvents {...props} />);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot(); 
    })

    it('good elements', () => {
        expect(wrapper.find('EventCard').length).toEqual(2);
        wrapper.setProps({
            events: {
                ...props.events,
                events: [],
                isLoading: true
            }
        });
        expect(wrapper.find('Loading').length).toEqual(1);
        wrapper.setProps({
            events: {
                ...props.events,
                events: [],
                error: 'error',
            }
        });
        expect(wrapper.find('#error').length).toEqual(1);
    })
})