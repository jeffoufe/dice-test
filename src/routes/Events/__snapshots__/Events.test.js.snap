// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events renders correctly 1`] = `
<Fragment>
  <Connect(withRouter(Header)) />
  <div
    className="events-container"
  >
    <EventCard
      apple_music_tracks={Array []}
      currency="GBP"
      date="date"
      description="description"
      event_images={
        Object {
          "landscape": "landscape",
          "square": "square",
        }
      }
      featured={true}
      key="event-card-0"
      lineup={Array []}
      location={
        Object {
          "city": "city",
          "country": "country",
        }
      }
      name="name"
      sale_start_date="2019-08-19"
      sold_out={false}
      spotify_tracks={Array []}
      ticket_types={
        Array [
          Object {
            "price": Object {
              "total": 800,
            },
          },
          Object {
            "price": Object {
              "total": 400,
            },
          },
        ]
      }
      url="url"
      venue="venue"
    />
    <EventCard
      apple_music_tracks={Array []}
      currency="GBP"
      date="date"
      description="description"
      event_images={
        Object {
          "landscape": "landscape",
          "square": "square",
        }
      }
      featured={false}
      key="event-card-1"
      lineup={Array []}
      location={
        Object {
          "city": "city",
          "country": "country",
        }
      }
      name="name"
      sale_start_date="2019-08-19"
      sold_out={false}
      spotify_tracks={Array []}
      ticket_types={
        Array [
          Object {
            "price": Object {
              "total": 800,
            },
          },
          Object {
            "price": Object {
              "total": 400,
            },
          },
        ]
      }
      url="url"
      venue="venue"
    />
  </div>
</Fragment>
`;
